helper/resource/state.go

```diff
@@ -128,6 +138,11 @@ func (conf *StateChangeConf) WaitForState() (interface{}, error) {
 						if conf.ContinuousTargetOccurence == targetOccurence {
 							return
 						} else {
+							// FIXME: I think this continue is buggy, it's continuing the for
```

Comment:
Yes, I'm aware of what the `continue` does. However, that's not my continue, that continue existed in the code prior to my changes. I'm simply flagging that I think there's a small bug that pre-existed in the code, and that if you wanted to fix it, you'd need to call `refreshMutex.Unlock()` (which goes back to what you're saying about this being somewhat easy to introduce bugs into, and I'm trying to guard against that).